rudder-license-cli/src/main/scala/com/normation/license/cli/LicenseCli.scala

```diff
@@ -0,0 +1,177 @@
+/*
+*************************************************************************************
+* Copyright 2017 Normation SAS
+*************************************************************************************
+*
+* This file is part of Rudder.
+*
+* Rudder is free software: you can redistribute it and/or modify
+* it under the terms of the GNU General Public License as published by
+* the Free Software Foundation, either version 3 of the License, or
+* (at your option) any later version.
+*
+* In accordance with the terms of section 7 (7. Additional Terms.) of
+* the GNU General Public License version 3, the copyright holders add
+* the following Additional permissions:
+* Notwithstanding to the terms of section 5 (5. Conveying Modified Source
+* Versions) and 6 (6. Conveying Non-Source Forms.) of the GNU General
+* Public License version 3, when you create a Related Module, this
+* Related Module is not considered as a part of the work and may be
+* distributed under the license agreement of your choice.
+* A "Related Module" means a set of sources files including their
+* documentation that, without modification of the Source Code, enables
+* supplementary functions or services in addition to those offered by
+* the Software.
+*
+* Rudder is distributed in the hope that it will be useful,
+* but WITHOUT ANY WARRANTY; without even the implied warranty of
+* MERCHANTABILITY or FITNESS FOR A PARTICULAR PURPOSE.  See the
+* GNU General Public License for more details.
+*
+* You should have received a copy of the GNU General Public License
+* along with Rudder.  If not, see <http://www.gnu.org/licenses/>.
+
+*
+*************************************************************************************
+*/
+
+package com.normation.license.cli
+
+import scopt.OptionParser
+import ca.mrvisser.sealerate.values
+import com.normation.license.MaybeLicenseError._
+import com.normation.license.LicenseSigner
+import com.normation.license.LicenseReader
+import com.normation.license.RudderLicense
+import com.normation.license.RSAKeyManagement
+import com.normation.license.LicenseChecker
+import com.normation.license.LicenseError
+
+/**
+ * The configuration object for our CLI.
+ * The basic process is to take one file in input for the definition of variables, one set of files as template to change.
+ *
+ * By default, the files are generated with the same name in the current folder.
+ *
+ * - add the possibility to take directories as input (but a good shell can do it, so not very important)
+ */
+
+sealed trait CliMode
+final object CliMode {
+  final case object Check extends CliMode
+  final case object Sign  extends CliMode
+
+  val all = values[CliMode]
+}
+
+final case class Config(
```

Comment:
So you define a value for "LicenseOutPath" even if you don't need it (ie Check) 

Should we not make it more demand on the CliMode ? either by integrating parameter into cliMode or changing the class model (make config a Trait with a generic CliMode, and two case class defining CheckConfig and SignConfig ? 

Maybe make licenceOutPath Optionnal ?

Reply:
It seems to be the way the cli arg parser is designed to work. It's not very satisfying, as you remarked. I will try&test to see if there is better model accepted by the lib.